src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  AriaAttributes,
+  ChangeEvent,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends Pick<AriaAttributes, "aria-label"> {
+  name?: string
+  placeholder?: string
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  debounceDelay?: number
+  initialValue?: string
+  onSearch?: (searchTerm: string) => void
+  showSearchButton?: boolean
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onFocus?: (e: FocusEvent<HTMLInputElement>) => void
+}
+
+export function SearchInput({
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onSearch,
+  debounceDelay = 300,
+  showSearchButton,
+  initialValue = "",
+  onBlur,
+  onFocus,
+  ...aria
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState("")
+  const [cancelButtonHidden, setCancelButtonHidden] = useState(true)
+
+  useEffect(() => {
+    setSearchTerm(initialValue)
+  }, [initialValue])
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout | undefined
+    if (searchTerm) {
+      timer = setTimeout(() => onSearch?.(searchTerm), debounceDelay)
+    }
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This looks a bit suspicious, if the intent is avoid the effect in case the callback changes (because it could be a new function created on the render of the parent component) I think the best practice is to save that to a `ref`

Reply:
Yeah, that's a good call. Not sure why I went that route given that I've used a ref everywhere else in this repo.